lib/gtf/gz_reader.cc

```diff
@@ -0,0 +1,75 @@
+/*
+Author: Hossein Asghari
+*/
+
+#include <string>
+#include <zlib.h>
+#include "gz_reader.h"
+
+gz_reader::gz_reader()
+{
+    gzinput = Z_NULL;
+    zbuffer = (char *) malloc(BUFFSIZE);
+
+    buff_pos = 0;
+    buff_size = 0;
+}
+
+gz_reader::~gz_reader() 
+{
```

Comment:
Let's go RAII all the way. Can you add a call to `close_gz()` in the destructor.